src/org/labkey/test/tests/SampleTypeLinkToStudyTest.java

```diff
@@ -357,6 +356,106 @@ public void testAssaySupport()
         new DataRegionTable("Data", getDriver()).clickHeaderButtonAndWait("Link to Study");
     }
 
+    @Test
+    public void testQCState()
```

Comment:
I'm curious why this test is here? It doesn't seem to have any relationship to sample types or link to study (other than it's a sample type dataset). There are other QC state related tests in `StudyTest.class`. Also, is there not an existing test that accomplishes the same scenario?

Reply:
This was the part i tested as Final Polish for Study.

Reply:
Perhaps the test could be pared down to retain the part where it checks the `Data linked to this study` option in Manage Dataset QC States is in fact set to the state indicated in the dropdown. The other part, changing the QC state of one row, could probably be omitted.